service/car_cleaner_service.rb

```diff
@@ -0,0 +1,114 @@
+require 'time'
+require_relative '../model/car_cleaner'
+require_relative '../model/car'
+
+class CarCleanerService
+
+  ONE_HOUR_IN_SECONDS = 3600
+
+  attr_reader :car_cleaner, :time
+
+  def initialize(time)
+    @car_cleaner = CarCleaner.new
+    @time = time
+  end
+
+  def add_car(car)
+    # if the client did not specify the time they need the car, I consider it to be the end of the day
+    car.required_time = Time.parse(time.strftime("%Y-%m-%dT#{today_closing_hour}:00:00%z")) if car.required_time.nil?
```

Comment:
you could remove this comment and do something like this, to be more intuitive and easier to understand:
```ruby
end_of_day = Time.parse(time.strftime("%Y-%m-%dT#{today_closing_hour}:00:00%z"))
car.required_time = end_of_day if car.required_time.nil?
```

Reply:
That's a good idea, thanks!